components/createPost/CreatePost.tsx

```diff
@@ -0,0 +1,107 @@
+  }
+
+  return (
+    <div className={createPostStyles.createPostContainer}>
+      <div className={createPostStyles.headingAndCloseButtonContainer}>
+        <h3>Create new post</h3>
+        <button className={buttonsStyles.closeIconButton}>
+          <FontAwesomeIcon icon={faClose} />
+        </button>
+      </div>
+      <div className={createPostStyles.postInputTextAreaContainer}>
+        <textarea
+          placeholder="Write something..."
+          maxLength={wordsLimitForPostTextInput}
+          className={createPostStyles.postInputTextArea}
+          onChange={(e) => setWordCount(e.target.value.length)}
+        />
+        <span className={createPostStyles.wordCounter}>
+          {wordCount}/{wordsLimitForPostTextInput}
+        </span>
+      </div>
+      <div className={createPostStyles.postCategoryDropdownContainer}>
+        <select>
+          <option value="general">General</option>
+          <option value="query">Query</option>
+        </select>
+      </div>
+      {uploadedImages && (
+        <div className={createPostStyles.uploadedImagesPreviewsContainer}>
+          {uploadedImages.map((file) => (
+            <div
+              key={file.name}
+              className={
+                createPostStyles.individualUploadedImagePreviewsContainer
+              }
+            >
+              <button
+                className={createPostStyles.uploadedImagePreviewRemoveButton}
+              >
+                <FontAwesomeIcon
+                  icon={faClose}
+                  onClick={() => removeImage(file)}
+                />
+              </button>
+              <img
+                className={createPostStyles.uploadedImagePreview}
+                src={URL.createObjectURL(file)}
+              />
+            </div>
+          ))}
+        </div>
+      )}
+      <div className={createPostStyles.utilitiesContainer}>
+        <div className={createPostStyles.iconButtonsContainer}>
+          <button>
+            <FontAwesomeIcon icon={faUserTag} />
+          </button>
+          <button>
+            <label
+              htmlFor="post-image"
+              className={createPostStyles.labelForPostImage}
+            >
+              <FontAwesomeIcon icon={faImage} />
+              <input
+                type="file"
+                accept="image/*"
+                id="post-image"
+                className={createPostStyles.inputForPostImage}
+                onChange={(e) => {
+                  console.log(e.target.files![0]);
+                  if (uploadedImages.length === 3) return;
+                  if (e.target.files === null) return;
+                  if (e.target.files.length === 0) return;
+                  if (isImage(e.target.files[e.target.files.length - 1])) {
+                    setUploadedImages((prev) => [
+                      ...prev,
+                      e.target.files![e.target.files!.length - 1],
+                    ]);
+                  }
+                }}
```

Comment:
Since there is a lot of code in onChange, you could have created a function for this. That would be more readable. Try to minimize writing long conditional code in JSX